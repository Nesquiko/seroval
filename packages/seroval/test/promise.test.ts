import { describe, it, expect } from 'vitest';
import {
  AsyncServerValue,
  Feature,
  serializeAsync,
  toJSONAsync,
} from '../src';

describe('Promise', () => {
  describe('compat', () => {
    it('should throw an error for unsupported target', async () => {
      await expect(() => serializeAsync(Promise.resolve('test'), {
        disabledFeatures: Feature.Promise,
      }))
        .rejects
        .toThrowErrorMatchingSnapshot();
    });
    it('should use function expression instead of arrow functions', async () => {
      const example: Record<string, AsyncServerValue> = {};
      example.self = Promise.resolve(example);
      expect(await serializeAsync(example, {
        disabledFeatures: Feature.ArrowFunction,
      }))
        .toMatchSnapshot();
    });
  });
  describe('compat#toJSON', () => {
    it('should throw an error for unsupported target', async () => {
      await expect(() => toJSONAsync(Promise.resolve('test'), {
        disabledFeatures: Feature.Promise,
      }))
        .rejects
        .toThrowErrorMatchingSnapshot();
    });
  });
});
