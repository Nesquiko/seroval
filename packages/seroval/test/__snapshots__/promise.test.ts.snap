// Vitest Snapshot v1

exports[`Promise > compat > should throw an error for unsupported target 1`] = `"Unsupported type \\"Promise\\""`;

exports[`Promise > compat > should use function expression instead of arrow functions 1`] = `"(function(h){return h={self:Promise.resolve().then(function(){return h})}})()"`;

exports[`Promise > compat#toJSON > should throw an error for unsupported target 1`] = `"Unsupported type \\"Promise\\""`;
